### PR TITLE
Feature/delimiter

### PR DESCRIPTION
**Pull request by assureclaims**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/ngx-localstorage/src/lib/interfaces/storage-configuration.ts [hunk 1]
```diff
@@ -14,4 +14,8 @@ export interface NgxLocalstorageConfiguration {
      * Determines the storage type. (Default: localStorage)
      */
     storage?: Storage
+    /**
+     * Determines the delimiter, will used befoure key. (Default: _)
+     */
+     delimiter?: string
```
Comment (bohoffi) [inline libs/ngx-localstorage/src/lib/interfaces/storage-configuration.ts:20]: When extending the configuration the extension should be reflected in the documentation.

Furthermore I'd suggest adding the default value `_` to the `defaultConfig` object exported in `utils.ts`

File: libs/ngx-localstorage/src/lib/utils.ts [hunk 2]
```diff
@@ -32,10 +32,11 @@ export const setProperty = (path: string[] | string, value: any, object: any, fa
 /**
  * Constructs the storage key based on a prefix - if given - and the key itself
  */
-export const constructKey = (key: string, prefix?: string, configuredPrefix?: string): string => {
+export const constructKey = (key: string, prefix?: string, configuredPrefix?: string, delimiter?: string): string => {
   const prefixToUse = prefix || configuredPrefix;
+  const delimietrToUse = delimiter || '_';
```
Comment (bohoffi) [inline libs/ngx-localstorage/src/lib/utils.ts]: When using a default value defined in the `defaultConfig` you don't have to handle it here and just pass it through

File: libs/ngx-localstorage/src/lib/services/promisable.service.ts [hunk 3]
```diff
@@ -16,6 +16,10 @@ export class PromisableService {
     private readonly configuration: NgxLocalstorageConfiguration,
     private readonly defaultSerializer: StorageSerializer
   ) {
+    if(this.configuration.delimiter == null || this.configuration.delimiter != null && this.configuration.delimiter == '')
```
Comment (bohoffi) [inline libs/ngx-localstorage/src/lib/services/promisable.service.ts]: WHen using the default value provided by `defaultConfig` there's no check or reassignment needed here

**Review by bohoffi — CHANGES_REQUESTED** [review]
What needs to be done:

- [x] extend `defaultConfig`
- [x] use config value for delimiter (remove checks and reassignments)
- [ ] update documentation

File: libs/ngx-localstorage/src/lib/interfaces/storage-configuration.ts (hunk 1 above)
Comment (assureclaims) [reply libs/ngx-localstorage/src/lib/interfaces/storage-configuration.ts:20]: Fixed

File: libs/ngx-localstorage/src/lib/utils.ts (hunk 2 above)
Comment (assureclaims) [reply libs/ngx-localstorage/src/lib/utils.ts]: Fixed

File: libs/ngx-localstorage/src/lib/services/promisable.service.ts (hunk 3 above)
Comment (assureclaims) [reply libs/ngx-localstorage/src/lib/services/promisable.service.ts]: Fixed

**Comment by assureclaims** [comment]
Thanks. Update the changes please review.

**Review by bohoffi — CHANGES_REQUESTED** [review]
Code changes are good now but the documentation update is still missing - which means updating the [README](https://github.com/bohoffi/ngx-localstorage#configuration-ngxlocalstoragemoduleforrootmoduleconfig)

**Comment by assureclaims** [comment]
I updated the document please review and update if any thing else is needed.

**Review by bohoffi — CHANGES_REQUESTED** [review]
May I request one more update from you?
While reviewing your changes using the demo application I've noticed that the decorator didn't got the stored value anymore - which is logical because the decorator don't use the configuration token.

- [ ] Extend `DecoratorOpts` with an optional `delimiter` property
- [ ] Updating the decorators function to init the delimter (analog to the prefix) and hand it to the service instances config
- [ ] Updating the `decorator.component` code and template (use the demo configurations delimiter)

**Comment by assureclaims** [comment]
> May I request one more update from you?
> While reviewing your changes using the demo application I've noticed that the decorator didn't got the stored value anymore - which is logical because the decorator don't use the configuration token.
> 
> * [ ]  Extend `DecoratorOpts` with an optional `delimiter` property
> * [ ]  Updating the decorators function to init the delimter (analog to the prefix) and hand it to the service instances config
> * [ ]  Updating the `decorator.component` code and template (use the demo configurations delimiter)

**If I will make that changes, so every one using local storage with serializer in there project will get impacted because its need to make changes in series of set function parameters. so if it's required please let me know I will make the changes.**

**Comment by assureclaims** [comment]
> Code changes are good now but the documentation update is still missing - which means updating the [README](https://github.com/bohoffi/ngx-localstorage#configuration-ngxlocalstoragemoduleforrootmoduleconfig)

**Comment by bohoffi** [comment]
@assureclaims I'll recheck the PR during the next days

**Comment by bohoffi** [comment]
I decided to not include the decorator changes for now so this will be approved

**Review by bohoffi — APPROVED** [review]

**Comment by assureclaims** [comment]
Thank you so much :)

**Pull request merged**